Examples/Rendering/ManyRenderWindows/index.js

```diff
@@ -117,42 +141,65 @@ function addRenderWindow() {
     vtkInteractorStyleTrackballCamera.newInstance()
   );
 
+  const button = createRemoveButton();
+  button.addEventListener('click', () => {
+    rootContainer.removeChild(container);
+    mainRenderWindow.removeRenderWindow(renderWindow);
+    mainRenderWindowView.removeNode(renderWindowView);
+    interactor.delete();
+    renderWindowView.delete();
+    renderWindow.delete();
+    if (mainRenderWindow.getChildRenderWindowsByReference().length === 0) {
+      // When there is no child render window anymore, delete the main render window
+      // We also release the graphics resources, which is not very import as when the context is destroyed, the resources should be freed
+      // The release of shared graphics resources is not automatic and can be done by hand when the resource is not needed anymore
+      const imageData = actor.getMapper().getInputData();
+      const scalars = imageData.getPointData().getScalars();
+      mainRenderWindowView.releaseGraphicsResourcesForObject(scalars);
+      mainRenderWindowView.delete();
+      mainRenderWindow.delete();
+    }
+  });
+  container.appendChild(button);
+
+  // Create the corresponding renderer
+  const background = [
+    0.5 * Math.random() + 0.25,
+    0.5 * Math.random() + 0.25,
+    0.5 * Math.random() + 0.25,
+  ];
+  const renderer = vtkRenderer.newInstance({ background });
+  renderWindow.addRenderer(renderer);
+
+  // Add the actor and reset camera
+  renderer.addActor(actor);
+  const camera = renderer.getActiveCamera();
+  camera.yaw(90);
+  camera.roll(90);
+  camera.azimuth(Math.random() * 360);
+  renderer.resetCamera();
+
+  childRenderWindows.push(renderWindow);
+
   return renderWindow;
 }
 
 // ----------------------------------------------------------------------------
 // Fill up page
 // ----------------------------------------------------------------------------
 
-const childRenderWindows = [];
 createActor().then((actor) => {
-  // Main view has to be initialized before the first "render" from a child render window
-  // We initialize before creating the child render windows because the interactor initialization calls "render" on them
-  mainRenderWindowView.initialize();
-
+  // The button to add render windows
+  addRenderWindowButton.addEventListener('click', () => {
+    addRenderWindow(actor);
+    mainRenderWindowView.resizeFromChildRenderWindows();
```

Comment:
shouldn't this go to addRenderWindow for simplicity ?

Reply:
The user can decide not to resize the main renderwindow until all children are added, or even set the size of the main renderwindow to something big enough with setSize() and never call resizeFromChildRenderWindows()